### ISSUE TITLE
Feature: there is no overload for `AndThenAnswer`

### ISSUE THREAD

**Issue by LeoColman** [mockk/mockk#343]
As discusse in slack, it's possible to do

```
every { foo() } returns "Bar" thenReturns "Bat"
```

but currently, impossible to do

```
every {  foo() } answers { "Bar" } andThenAnswer { "Bat" } 
```

as this overload doesn't exist.

Curiously,

```
every { foo() } answers { "Bar" } andThen { "Bat" }
``` 

works.

It seems to be a naming issue. We should deprecate this version and add the correct overload

**Comment by Kerooker** [comment]
unstale

**Comment by Raibaz** [comment]
This was fixed by #528
